### ISSUE TITLE
Docs & Examples for Multi-Fidelity / Early Stopping

### ISSUE THREAD

**Issue by bbudescu** [PKU-DAIR/open-box#51]
E.g., is the resource dimension (e.g., the number of epochs or the number of samples to train) treated as just another hyperparameter? How can one do cross-task transfer learning with multi-fidelity, e.g., how does one report xval accuracy at every epoch for previous trials?

**Comment by bbudescu** [comment]
Sorry for so many questions over such little time. I'm trying to decide which optimization package to use for a project, and, for that, I'm checking the support for the features I need / would like.

**Comment by jhj0411jhj** [comment]
For early stopping, user may implement the strategy during optimization via the [ask-and-tell interface](https://github.com/PKU-DAIR/open-box/blob/master/examples/ask_and_tell_interface.py). We will soon update the API of `Optimizer` to support user-defined early stoping functions.
